shared_model/validators/block_validator.hpp

```diff
@@ -37,12 +40,48 @@ namespace shared_model {
        * @return Answer containing found error if any
        */
       Answer validate(const interface::Block &block) const override {
-        return ContainerValidator<interface::Block,
-                                  FieldValidator,
-                                  TransactionsCollectionValidator>::
+        Answer answer = ContainerValidator<interface::Block,
+                                           FieldValidator,
+                                           TransactionsCollectionValidator>::
             validate(block, "Block", [this](auto &reason, const auto &cont) {
               this->field_validator_.validateHash(reason, cont.prevHash());
             });
+
+        validation::ReasonsGroupType block_reason;
+        std::unordered_set<std::string> hashes = {};
+
+        auto rejected_hashes = block.rejected_transactions_hashes();
+
+        for (const auto &hash : rejected_hashes | boost::adaptors::indexed(0)) {
+          if (hashes.count(hash.value().hex())) {
+            block_reason.second.emplace_back(
+                (boost::format("Rejected hash '%s' with index "
+                               "'%d' has already appeared in a block")
+                 % hash.value().hex() % hash.index())
+                    .str());
+          } else {
+            hashes.insert(hash.value().hex());
+          }
+        }
+
+        auto transaction = block.transactions();
+
+        for (const auto &tx : transaction | boost::adaptors::indexed(0)) {
+          auto hex = tx.value().hash().hex();
+          if (hashes.count(hex)) {
+            block_reason.second.emplace_back(
+                (boost::format("Hash '%s' of transaction "
+                               "'%d' has in rejected hash")
```

Comment:
```suggestion
                               "'%d' has already appeared in rejected hashes")
```
Maybe? Because "has in rejected hash" does not sound right to me.

Reply:
Fixed